lib/vericite.rb

```diff
@@ -281,13 +281,18 @@ def sendRequest(command, args)
           context_id = course.id
           assignment_id = assignment.id
           user_id = user.id
-          user_score_cache_key = "vericite_scores/#{consumer}/#{context_id}/#{user_id}"
-          # first check if the cache already has the score:
-          user_score_map = Rails.cache.read(user_score_cache_key)
-          if user_score_map == nil
+          user_score_cache_key_prefix =  "vericite_scores/#{consumer}/#{context_id}/#{assignment_id}/"
+          users_score_map = {}
+          # first check if the cache already has the user's score and if we haven't looked up this assignment lately:
+          users_score_map["#{user_id}"] = Rails.cache.read("#{user_score_cache_key_prefix}#{user_id}")
+          if users_score_map["#{user_id}"] == nil && Rails.cache.read(user_score_cache_key_prefix) == nil
+            #we already looked up this user in Redis, don't bother again (by setting {})
+            users_score_map["#{user_id}"] ||= {}
             # we need to look up the user scores in VeriCite for this course
             # @return [Array<ReportScoreReponse>]
-            data, status_code, headers = vericite_client.reports_scores_context_id_get(context_id, consumer, consumer_secret, {:'user_id' => user_id})
+            data, status_code, headers = vericite_client.reports_scores_context_id_get(context_id, consumer, consumer_secret, {:'assignment_id' => assignment_id})
```

Comment:
Either just `:assignment_id => assignment_id` or `assignment_id: assignment_id`  The later is the preferred form these days.